.ci/scripts/process_metric_data.py

```diff
@@ -36,27 +41,214 @@ def remove_unwanted_lines(csv_file):
+
+    Parameters
+    ----------
+    stats_row : dictionary
+        row from data collection of docker stats
 
+    Returns : string
+    -------
+        formatted, comma separated string
 
-def main(csv_file):
+    """
+
+    secs = stats_row['SECONDS']
+    name = stats_row['{{.Name}}']
+    pids = stats_row['{{.PIDs}}']
+    cpu = stats_row['{{.CPUPerc}}'].replace("%", "")
+    mem = get_mem_gb(stats_row['{{.MemUsage}}'])
+    mem_p = stats_row['{{.MemPerc}}'].replace("%", "")
+    net_s, net_r = get_net_send_recv(stats_row['{{.NetIO}}'])
+    disk_r, disk_w = get_disk_read_write(stats_row['{{.BlockIO}}'])
+    return f"{secs},{name},{pids},{cpu},{mem},{mem_p},{net_s},{net_r},{disk_r},{disk_w}"
+
+
+def make_lists(csv_file):
+    """
+    Makes a list of dictionaries for each row of the csv file captured during the docker run
+
+    Parameters
+    ----------
+    csv_file : file
+        file used to capture either docker stats or miscellaneous OS measurements
+
+    Returns
+    -------
+    csv_to_list : list
+        List that is now ready to be formatted
+
+    """
+    csv_to_list = []
+    with open(csv_file) as csv_handle:
+        reader = csv.DictReader(csv_handle)
+        for row in reader:
+            csv_to_list.append(row)
+    return csv_to_list
+
+
+def main():
     """ main program in process_metric_data.py"""
-    remove_unwanted_lines(csv_file)
-    delete_temp_files(csv_file)
+
+    container_info = sys.argv[1]
+    stats_file = sys.argv[2]
+    misc_file = sys.argv[3]
+
+    temp_stats = "temp_opera_docker_stats.csv"
+    temp_misc = "temp_opera_misc_stats.csv"
+
+    # Remove lines that may have been recorded before Docker stated.
+    remove_unwanted_lines(stats_file, temp_stats)
+    remove_unwanted_lines(misc_file, temp_misc)
+
+    current_time = datetime.datetime.today().strftime('%Y%m%d_%H%M%S')
+
+    # For now make two formatted files
+    docker_report_file = f"docker_metrics_{container_info}_{current_time}.csv"
+    misc_report_file = f"misc_metrics_{container_info}_{current_time}.csv"
+
+    # read files into lists
+    stats_list = make_lists(temp_stats)
+    misc_list = make_lists(temp_misc)
+
+    # Write out the docker stats file
+    docker_columns = "SECONDS, Name, PIDs, CPU, Memory, MemoryP, NetSend, NetRecv, DiskRead, DiskWrite"
+    # Todo - used to convert non-string data to numbers
+    # convert = {'SECONDS': int(), 'Name': str(), 'PIDs': int(), 'CPU': float(), 'Memory': float(), 'MemoryP': float(),
+    #            'NetSend': float(), 'NetRecv': float(), 'DiskRead': float(), 'DiskWrite': float()}
+    with open(docker_report_file, 'w') as out_file:
+        out_file.write(f"{docker_columns}\n")
+        for stats_row in stats_list:
+            row = format_out_row_docker(stats_row)
+            out_file.write(f"{row}\n")
+
+    # Write out the miscellaneous file
+    # Todo - divide out mac OS runs and linux runs - Swap will work on linux
```

Comment:
See my comment in the shell script about keeping a column for swap space even if it is not used.

Reply:
Done